core/client/fs/src/main/java/alluxio/client/block/stream/NettyPacketWriter.java

```diff
@@ -498,6 +526,33 @@ public void operationComplete(ChannelFuture future) {
           mDoneOrFailed.signal();
           mBufferNotFullOrFailed.signal();
           mBufferEmptyOrFailed.signal();
+          mFlushedOrFailed.signal();
+        }
+      }
+    }
+  }
+
+  /**
+   * The netty channel future listener that is called when a FLUSH is complete.
+   */
+  private final class FlushListener implements ChannelFutureListener {
```

Comment:
Contrary to my previous comment, I think we can reuse the `EofOrCancelListener`.